File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.py
class Solution(object):
    def lengthOfLongestSubstring(self, s):
        """
        :type s: str
        :rtype: int
        """
        hashSet=set()
        left=0
        maxlength=0

        for right in range(len(s)):
            while s[right] in hashSet:
                hashSet.remove(s[left])
                left+=1
            hashSet.add(s[right])
           
            maxlength=max(maxlength, right-left+1)
        return maxlength

        

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        unordered_set<char> set;
        int maxlen=0;
        int left =0;
        for (int right =0;right<s.length();right++){
            while(set.contains(s[right])){
                set.erase(s[left]);
                left++;
            }
            set.insert(s[right]);
            maxlen= max(maxlen, right-left+1);
            
        }
        return maxlen;
    }
};